x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/endpoint_activity_log.tsx

```diff
@@ -19,54 +27,95 @@ import {
   getActivityLogError,
   getActivityLogIterableData,
   getActivityLogRequestLoaded,
+  getLastLoadedActivityLogData,
   getActivityLogRequestLoading,
 } from '../../store/selectors';
 
+const Sentinel = styled.div`
+  height: 6px;
+  width: 100%;
+`;
+
 export const EndpointActivityLog = memo(
   ({ activityLog }: { activityLog: AsyncResourceState<Immutable<ActivityLog>> }) => {
     const activityLogLoading = useEndpointSelector(getActivityLogRequestLoading);
     const activityLogLoaded = useEndpointSelector(getActivityLogRequestLoaded);
+    const activityLastLogData = useEndpointSelector(getLastLoadedActivityLogData);
     const activityLogData = useEndpointSelector(getActivityLogIterableData);
+    const activityLogSize = activityLogData.length;
     const activityLogError = useEndpointSelector(getActivityLogError);
-    const dispatch = useDispatch<(a: EndpointAction) => void>();
-    const { page, pageSize } = useEndpointSelector(getActivityLogDataPaging);
+    const dispatch = useDispatch<(action: EndpointAction) => void>();
+    const { page, pageSize, disabled: isPagingDisabled } = useEndpointSelector(
+      getActivityLogDataPaging
+    );
+
+    const fetchMoreSentinel = useRef<HTMLInputElement | null>(null);
+    const getActivityLog = useCallback(
+      (entries: IntersectionObserverEntry[]) => {
+        const isIntersecting = entries.some((entry) => entry.intersectionRatio > 0);
+        if (isIntersecting && activityLogLoaded && !isPagingDisabled) {
+          dispatch({
```

Comment:
Q: Why you don't need to dispatch `endpointDetailsActivityLogUpdatePaging` here?

Reply:
Yeah, that's because here we're only fetching new data and this should happen every time we scroll to the bottom. We dispatch `endpointDetailsActivityLogUpdatePaging ` when we want to disable paging and backtrack the page number when there's no data after fetch.